src/main/cpp/src/ftos_converter.cuh

```diff
@@ -1212,48 +1212,55 @@ __device__ inline T round_half_even(T const input, int const olength, int const
   return num;
 }
 
-__device__ inline int to_formated_double_chars(floating_decimal_64 const v,
-                                               bool const sign,
-                                               char* const result,
-                                               int digits)
+/*
+ * Convert a floating_decimal_32/64 to a formatted string as the default format (#,###,###.##)
+ * of format_number in Spark.
+ *
+ * @param v: floating_decimal_32/64 value
+ * @param sign: sign of the number
+ * @param result: output string
+ * @param digits: number of digits after decimal point
+ */
+template <typename T, typename U>
```

Comment:
Since `U` depends on `T`, we can just remove it here, then:
```
__device__ to_formatted_chars(...) {
static_assert(std::is_same_v<T, floating_decimal_32> || std::is_same_v<T, floating_decimal_64>);
using U = std::conditional_t<std::is_same_v<T, floating_decimal_32>, uint32_t, uint64_t>;

Reply:
The same can be applied to the function `format_size` below.

Reply:
done.